src/lib/logical_query_plan/mock_node.cpp

```diff
@@ -123,6 +129,26 @@ UniqueColumnCombinations MockNode::unique_column_combinations() const {
   return unique_column_combinations;
 }
 
+void MockNode::set_order_constraints(const TableOrderConstraints& order_constraints) {
+  _order_constraints = order_constraints;
+}
+
+OrderDependencies MockNode::order_dependencies() const {
+  auto order_dependencies = OrderDependencies{};
+  for (const auto& order_constraint : _order_constraints) {
+    const auto& ordering_columns = order_constraint.ordering_columns();
+    const auto& ordered_columns = order_constraint.ordered_columns();
+    if (contains_any_column(_pruned_column_ids, ordering_columns) ||
+        contains_any_column(_pruned_column_ids, ordered_columns)) {
```

Comment:
Wouldn't  `[a,b] -> [c,d,e]` still work if `e` is pruned?

Reply:
We would have to transform to `[a, b] |-> [c, d]`. However, it only really works for this exact case and we ignore that for now.